File: src/java/WebNinja/src/test/java/webninja/workflow/UserWorkflow.java
package webninja.workflow;

import org.joda.time.DateTime;
import webninja.selenuim.CodeTrack;
import webninja.selenuim.LogonPage;
import webninja.selenuim.ProjectForm;

public class UserWorkflow {
    private final ProjectWorkflow projectWorkflow;
    private final CodeTrack codeTrack;

    public UserWorkflow(CodeTrack codeTrack) {
        this.codeTrack = codeTrack;
        this.projectWorkflow = new ProjectWorkflow(codeTrack);
    }

    public UserWorkflow logOnAs(String user) {
        logOut();
        LogonPage logonPage = codeTrack.gotoLogonPage();
        logonPage.setName(user);
        logonPage.setPassword(user);
        codeTrack.submit();
        return this;
    }

    public ProjectWorkflow createNewProject() {
        ProjectForm projectForm = codeTrack.showAdminPage().showNewProjectForm();
        String nextProjectName = getNextProjectName();
        projectForm.setName(nextProjectName);
        projectForm.setDescription("Test project");
        codeTrack.submit();
        codeTrack.gotoHomePage();
        return usingProject(nextProjectName);
    }

    public ProjectWorkflow usingProject(String project) {
        projectWorkflow.setCurrentProject(project);
        return projectWorkflow;
    }


    public UserWorkflow logOut() {
        if (codeTrack.isLoggedIn())
            codeTrack.logOut();
        return this;
    }

    private String getNextProjectName() {
        return "CP" + new DateTime().toString("yyyyMMddhhmmss");
    }
}

File: src/dotnet/WebNinja/WebNinja/workflow/UserWorkflow.cs
using System;
using WebNinja.selenium;

namespace WebNinja.workflow
{
    public class UserWorkflow
    {
        private readonly CodeTrack _codeTrack;
        private readonly ProjectWorkflow _projectWorkflow;
        
        public UserWorkflow(CodeTrack codeTrack)
        {
            _codeTrack = codeTrack;
            _projectWorkflow = new ProjectWorkflow(_codeTrack);
        }

        public UserWorkflow LogonAs(string user)
        {
            Logout();
            LogonPage logonPage = _codeTrack.GotoLogonPage();
            logonPage.Name = user;
            logonPage.Password = user;
            logonPage.Submit();
            return this;
        }

        public ProjectWorkflow CreateNewProject()
        {
            ProjectForm projectForm = _codeTrack.GotoAdminPage().GotoProjectForm();
            string projectName = NextProjectName();
            projectForm.Name = projectName;
            projectForm.Description = "Test Project";
            projectForm.Submit();
            _codeTrack.GotoHomePage();
            return UsingProject(projectName);
        }

        private ProjectWorkflow UsingProject(string name)
        {
            _projectWorkflow.CurrentProject = name;
            return _projectWorkflow;
        }

        private void Logout()
        {
            if (_codeTrack.IsLoggedIn)
                _codeTrack.Logout();
        }

        private static String NextProjectName()
        {
            return "CP" + new DateTime().ToString("yyyyMMddhhmmss");
        }

    }
}